frontend/src/components/wallet/WalletConnectionStatus.tsx

```diff
@@ -0,0 +1,142 @@
+import { useState, useEffect, useContext } from 'react';
+import { utils } from 'ethers';
+import { WalletContext } from 'lib/wallet/WalletContext';
+import metamaskLogo from 'assets/img/metamask.svg';
+
+declare const window: Window &
```

Comment:
We should probably put this in a `/types` directory in the root of the project.

Reply:
I agree about have a central types directory.  This type definition is specific to this one component - it allows checking for `window.ethereum` to check for metamask presence.  I think it makes more sense to leave this one in the same file as the component.